src/System.Management.Automation/engine/Utils.cs

```diff
@@ -1153,8 +1153,22 @@ internal static bool PathIsUnc(string path)
 
         internal class NativeMethods
         {
-            [DllImport(PinvokeDllNames.GetFileAttributesDllName, CharSet = CharSet.Unicode, SetLastError = true)]
-            internal static extern int GetFileAttributes(string lpFileName);
+            private static string EnsureLongPathPrefix(string path)
+            {
+                if (!path.StartsWith(@"\\?\", StringComparison.Ordinal))
```

Comment:
Can we skip adding the prefix when the path is short?

Always adding the prefix will create a lot of garbage I think.